### PR TITLE
CORTX-32909: Update minor CORTX-Deployment.md Instructions with locally generated images support

### PR DESCRIPTION
**Pull request by mukul-seagate11**
Signed-off-by: Mukul Malhotra <mukul.malhotra@seagate.com>

# Problem Statement
- Changes added to update CORTX-Deployment.md with command to override & avoid the parameter to pull the images from GHCR

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM
- [ ] Jenkins pipelines used for testing

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [x] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

-----
[View rendered solutions/community-deploy/CORTX-Deployment.md](https://github.com/Seagate/cortx-re/blob/mukul-multinode-deployment/solutions/community-deploy/CORTX-Deployment.md)

### CODE REVIEW

**Comment by mukul-seagate11** [comment]
CORTX images after installing K8s cluster on cortx:

[root@ip-172-31-37-135 kubernetes]# docker images
REPOSITORY                                                                TAG              IMAGE ID       CREATED          SIZE
ip-172-31-37-135.ap-south-1.compute.internal:8080/seagate/cortx-rgw       2.0.0-0          4b3f9afdfa73   34 minutes ago   1.47GB
cortx-rgw                                                                 2.0.0-0          4b3f9afdfa73   34 minutes ago   1.47GB
cortx-all                                                                 2.0.0-0          cb2b23d7f892   35 minutes ago   1.19GB
ip-172-31-37-135.ap-south-1.compute.internal:8080/seagate/cortx-all       2.0.0-0          cb2b23d7f892   35 minutes ago   1.19GB
cortx-data                                                                2.0.0-0          51b703795ee6   35 minutes ago   1.09GB
ip-172-31-37-135.ap-south-1.compute.internal:8080/seagate/cortx-data      2.0.0-0          51b703795ee6   35 minutes ago   1.09GB
ip-172-31-37-135.ap-south-1.compute.internal:8080/seagate/cortx-control   2.0.0-0          e2ee07efffa7   36 minutes ago   842MB
cortx-control                                                             2.0.0-0          e2ee07efffa7   36 minutes ago   842MB
registry                                                                  2                3a0f7b0a13ef   2 days ago       24.1MB
nginx                                                                     latest           b692a91e4e15   10 days ago      142MB
calico/typha                                                              v3.23.3          beba684bccfb   3 weeks ago      122MB
calico/dikastes                                                           v3.23.3          4b1759fe9655   3 weeks ago      95.8MB
calico/flannel-migration-controller                                       v3.23.3          3f6e38b4def6   3 weeks ago      174MB
calico/kube-controllers                                                   v3.23.3          32d39d8db456   3 weeks ago      127MB
calico/cni                                                                v3.23.3          ecf96bae0aa7   3 weeks ago      254MB
calico/pod2daemon-flexvol                                                 v3.23.3          7d53150dd31a   3 weeks ago      18.8MB
calico/node                                                               v3.23.3          5f5175f39b19   3 weeks ago      203MB
k8s.gcr.io/kube-apiserver                                                 v1.22.12         d922296a8b85   4 weeks ago      128MB
k8s.gcr.io/kube-controller-manager                                        v1.22.12         f5ef4d6e36ee   4 weeks ago      122MB
k8s.gcr.io/kube-proxy                                                     v1.22.12         761817698b62   4 weeks ago      104MB
k8s.gcr.io/kube-scheduler                                                 v1.22.12         5ee90014579c   4 weeks ago      52.7MB
ghcr.io/seagate/cortx-build                                               rockylinux-8.4   d728841607b0   7 weeks ago      1.51GB
k8s.gcr.io/etcd                                                           3.5.0-0          004811815584   14 months ago    295MB
k8s.gcr.io/coredns/coredns                                                v1.8.4           8d147537fb7d   14 months ago    47.6MB
k8s.gcr.io/pause                                                          3.5              ed210e3e4a5b   17 months ago    683kB

**Comment by mukul-seagate11** [comment]
loaded cortx build images to worker nodes,

[root@ip-172-31-39-22 tmp]# for worker in $AWS_WORKER_IP;do ssh -i cortx.pem -o 'StrictHostKeyChecking=no' centos@$worker 'sudo docker load -i /tmp/cortximages.tar'; done
Loaded image: registry:2
Loaded image: nginx:latest
Loaded image: ghcr.io/seagate/cortx-build:rockylinux-8.4
Loaded image: cortx-data:2.0.0-0
Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-data:2.0.0-0
Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-all:2.0.0-0
Loaded image: cortx-all:2.0.0-0
Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-rgw:2.0.0-0
Loaded image: cortx-rgw:2.0.0-0
Loaded image: cortx-control:2.0.0-0
Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-control:2.0.0-0
Loaded image: registry:2
Loaded image: nginx:latest
Loaded image: ghcr.io/seagate/cortx-build:rockylinux-8.4
Loaded image: cortx-data:2.0.0-0
Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-data:2.0.0-0
Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-all:2.0.0-0
Loaded image: cortx-all:2.0.0-0
Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-rgw:2.0.0-0
Loaded image: cortx-rgw:2.0.0-0
Loaded image: cortx-control:2.0.0-0
Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-control:2.0.0-0

**Comment by mukul-seagate11** [comment]
Here is the output logs of deploying cortx stack as,

==================== Image Details ====================

ghcr.io/seagate/consul:1.11.4
ip-172-31-43-44.ap-south-1.compute.internal:8080/seagate/cortx-control:2.0.0-0
ip-172-31-43-44.ap-south-1.compute.internal:8080/seagate/cortx-data:2.0.0-0
ip-172-31-43-44.ap-south-1.compute.internal:8080/seagate/cortx-control:2.0.0-0
ghcr.io/seagate/kafka:3.0.0-debian-10-r97
ip-172-31-43-44.ap-south-1.compute.internal:8080/seagate/cortx-rgw:2.0.0-0
ghcr.io/seagate/zookeeper:3.8.0-debian-10-r9

==================== POD Status ====================

NAME                           READY   STATUS    RESTARTS      AGE     IP               NODE                                           NOMINATED NODE   READINESS GATES
cortx-consul-client-29xhl      1/1     Running   0             2m48s   192.168.195.10   ip-172-31-43-44.ap-south-1.compute.internal    <none>           <none>
cortx-consul-client-d26vw      1/1     Running   0             2m48s   192.168.143.66   ip-172-31-32-13.ap-south-1.compute.internal    <none>           <none>
cortx-consul-client-t6nzv      1/1     Running   0             2m48s   192.168.94.129   ip-172-31-33-104.ap-south-1.compute.internal   <none>           <none>
cortx-consul-server-0          1/1     Running   0             2m48s   192.168.94.133   ip-172-31-33-104.ap-south-1.compute.internal   <none>           <none>
cortx-consul-server-1          1/1     Running   0             2m48s   192.168.143.77   ip-172-31-32-13.ap-south-1.compute.internal    <none>           <none>
cortx-consul-server-2          1/1     Running   0             2m47s   192.168.195.14   ip-172-31-43-44.ap-south-1.compute.internal    <none>           <none>
cortx-control-7f8b78b8-mx6hg   1/1     Running   0             2m48s   192.168.143.82   ip-172-31-32-13.ap-south-1.compute.internal    <none>           <none>
cortx-data-g0-0                3/3     Running   0             2m46s   192.168.195.18   ip-172-31-43-44.ap-south-1.compute.internal    <none>           <none>
cortx-data-g0-1                3/3     Running   0             2m45s   192.168.143.79   ip-172-31-32-13.ap-south-1.compute.internal    <none>           <none>
cortx-data-g0-2                3/3     Running   0             2m45s   192.168.94.141   ip-172-31-33-104.ap-south-1.compute.internal   <none>           <none>
cortx-data-g1-0                3/3     Running   0             2m47s   192.168.143.68   ip-172-31-32-13.ap-south-1.compute.internal    <none>           <none>
cortx-data-g1-1                3/3     Running   0             2m46s   192.168.195.17   ip-172-31-43-44.ap-south-1.compute.internal    <none>           <none>
cortx-data-g1-2                3/3     Running   0             2m45s   192.168.94.135   ip-172-31-33-104.ap-south-1.compute.internal   <none>           <none>
cortx-ha-786f988667-2p8jp      3/3     Running   1 (35s ago)   2m48s   192.168.143.72   ip-172-31-32-13.ap-south-1.compute.internal    <none>           <none>
cortx-kafka-0                  1/1     Running   0             2m48s   192.168.94.134   ip-172-31-33-104.ap-south-1.compute.internal   <none>           <none>
cortx-kafka-1                  1/1     Running   0             2m48s   192.168.143.81   ip-172-31-32-13.ap-south-1.compute.internal    <none>           <none>
cortx-kafka-2                  1/1     Running   0             2m47s   192.168.195.21   ip-172-31-43-44.ap-south-1.compute.internal    <none>           <none>
cortx-server-0                 2/2     Running   0             2m48s   192.168.94.137   ip-172-31-33-104.ap-south-1.compute.internal   <none>           <none>
cortx-server-1                 2/2     Running   0             2m48s   192.168.195.12   ip-172-31-43-44.ap-south-1.compute.internal    <none>           <none>
cortx-server-2                 2/2     Running   0             2m47s   192.168.143.73   ip-172-31-32-13.ap-south-1.compute.internal    <none>           <none>
cortx-zookeeper-0              1/1     Running   0             2m48s   192.168.94.140   ip-172-31-33-104.ap-south-1.compute.internal   <none>           <none>
cortx-zookeeper-1              1/1     Running   0             2m48s   192.168.195.22   ip-172-31-43-44.ap-south-1.compute.internal    <none>           <none>
cortx-zookeeper-2              1/1     Running   0             2m47s   192.168.143.74   ip-172-31-32-13.ap-south-1.compute.internal    <none>           <none>

--------------- Sleeping for 1min before checking hctl status.... ---------------


--------------- hctl status ---------------

Sun Aug 14 09:00:53 UTC 2022
Unable to use a TTY - input is not a terminal or the right kind of file
Unable to use a TTY - input is not a terminal or the right kind of file
Unable to use a TTY - input is not a terminal or the right kind of file
Bytecount:
    critical : 0
    damaged : 0
    degraded : 0
    healthy : 0
Data pool:
    # fid name
    0x6f00000000000001:0x0 'storage-set-1__sns'
Profile:
    # fid name: pool(s)
    0x7000000000000001:0x0 'Profile_the_pool': 'storage-set-1__sns' 'storage-set-1__dix' None
Services:
    cortx-data-g0-0.cortx-data-headless.default.svc.cluster.local
    [started]  hax                 0x7200000000000001:0x0          inet:tcp:cortx-data-g0-0.cortx-data-headless.default.svc.cluster.local@22001
    [started]  ioservice           0x7200000000000001:0x1          inet:tcp:cortx-data-g0-0.cortx-data-headless.default.svc.cluster.local@21001
    [started]  confd               0x7200000000000001:0x2          inet:tcp:cortx-data-g0-0.cortx-data-headless.default.svc.cluster.local@21002
    cortx-data-g0-1.cortx-data-headless.default.svc.cluster.local
    [started]  hax                 0x7200000000000001:0x3          inet:tcp:cortx-data-g0-1.cortx-data-headless.default.svc.cluster.local@22001
    [started]  ioservice           0x7200000000000001:0x4          inet:tcp:cortx-data-g0-1.cortx-data-headless.default.svc.cluster.local@21001
    [started]  confd               0x7200000000000001:0x5          inet:tcp:cortx-data-g0-1.cortx-data-headless.default.svc.cluster.local@21002
    cortx-data-g0-2.cortx-data-headless.default.svc.cluster.local
    [started]  hax                 0x7200000000000001:0x6          inet:tcp:cortx-data-g0-2.cortx-data-headless.default.svc.cluster.local@22001
    [started]  ioservice           0x7200000000000001:0x7          inet:tcp:cortx-data-g0-2.cortx-data-headless.default.svc.cluster.local@21001
    [started]  confd               0x7200000000000001:0x8          inet:tcp:cortx-data-g0-2.cortx-data-headless.default.svc.cluster.local@21002
    cortx-data-g1-0.cortx-data-headless.default.svc.cluster.local  (RC)
    [started]  hax                 0x7200000000000001:0x9          inet:tcp:cortx-data-g1-0.cortx-data-headless.default.svc.cluster.local@22001
    [started]  ioservice           0x7200000000000001:0xa          inet:tcp:cortx-data-g1-0.cortx-data-headless.default.svc.cluster.local@21001
    [started]  confd               0x7200000000000001:0xb          inet:tcp:cortx-data-g1-0.cortx-data-headless.default.svc.cluster.local@21002
    cortx-data-g1-1.cortx-data-headless.default.svc.cluster.local
    [started]  hax                 0x7200000000000001:0xc          inet:tcp:cortx-data-g1-1.cortx-data-headless.default.svc.cluster.local@22001
    [started]  ioservice           0x7200000000000001:0xd          inet:tcp:cortx-data-g1-1.cortx-data-headless.default.svc.cluster.local@21001
    [started]  confd               0x7200000000000001:0xe          inet:tcp:cortx-data-g1-1.cortx-data-headless.default.svc.cluster.local@21002
    cortx-data-g1-2.cortx-data-headless.default.svc.cluster.local
    [started]  hax                 0x7200000000000001:0xf          inet:tcp:cortx-data-g1-2.cortx-data-headless.default.svc.cluster.local@22001
    [started]  ioservice           0x7200000000000001:0x10         inet:tcp:cortx-data-g1-2.cortx-data-headless.default.svc.cluster.local@21001
    [started]  confd               0x7200000000000001:0x11         inet:tcp:cortx-data-g1-2.cortx-data-headless.default.svc.cluster.local@21002
    cortx-server-0.cortx-server-headless.default.svc.cluster.local
    [started]  hax                 0x7200000000000001:0x12         inet:tcp:cortx-server-0.cortx-server-headless.default.svc.cluster.local@22001
    [started]  rgw_s3              0x7200000000000001:0x13         inet:tcp:cortx-server-0.cortx-server-headless.default.svc.cluster.local@22501
    cortx-server-1.cortx-server-headless.default.svc.cluster.local
    [started]  hax                 0x7200000000000001:0x14         inet:tcp:cortx-server-1.cortx-server-headless.default.svc.cluster.local@22001
    [started]  rgw_s3              0x7200000000000001:0x15         inet:tcp:cortx-server-1.cortx-server-headless.default.svc.cluster.local@22501
    cortx-server-2.cortx-server-headless.default.svc.cluster.local
    [started]  hax                 0x7200000000000001:0x16         inet:tcp:cortx-server-2.cortx-server-headless.default.svc.cluster.local@22001
    [started]  rgw_s3              0x7200000000000001:0x17         inet:tcp:cortx-server-2.cortx-server-headless.default.svc.cluster.local@22501
    
###############################################################################
                        Successfully Passed IO Sanity Testing
###############################################################################
--------------- SUCCESS ---------------

**Comment by shailesh-vaidya** [comment]
> loaded cortx build images to worker nodes,
> 
> [root@ip-172-31-39-22 tmp]# for worker in $AWS_WORKER_IP;do ssh -i cortx.pem -o 'StrictHostKeyChecking=no' centos@$worker 'sudo docker load -i /tmp/cortximages.tar'; done Loaded image: registry:2 Loaded image: nginx:latest Loaded image: ghcr.io/seagate/cortx-build:rockylinux-8.4 Loaded image: cortx-data:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-data:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-all:2.0.0-0 Loaded image: cortx-all:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-rgw:2.0.0-0 Loaded image: cortx-rgw:2.0.0-0 Loaded image: cortx-control:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-control:2.0.0-0 Loaded image: registry:2 Loaded image: nginx:latest Loaded image: ghcr.io/seagate/cortx-build:rockylinux-8.4 Loaded image: cortx-data:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-data:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-all:2.0.0-0 Loaded image: cortx-all:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-rgw:2.0.0-0 Loaded image: cortx-rgw:2.0.0-0 Loaded image: cortx-control:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-control:2.0.0-0

@mukul-seagate11 Seems you are not using the local registry framework but copying images manually within the server.

**Comment by mukul-seagate11** [comment]
> > loaded cortx build images to worker nodes,
> > [root@ip-172-31-39-22 tmp]# for worker in $AWS_WORKER_IP;do ssh -i cortx.pem -o 'StrictHostKeyChecking=no' centos@$worker 'sudo docker load -i /tmp/cortximages.tar'; done Loaded image: registry:2 Loaded image: nginx:latest Loaded image: ghcr.io/seagate/cortx-build:rockylinux-8.4 Loaded image: cortx-data:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-data:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-all:2.0.0-0 Loaded image: cortx-all:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-rgw:2.0.0-0 Loaded image: cortx-rgw:2.0.0-0 Loaded image: cortx-control:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-control:2.0.0-0 Loaded image: registry:2 Loaded image: nginx:latest Loaded image: ghcr.io/seagate/cortx-build:rockylinux-8.4 Loaded image: cortx-data:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-data:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-all:2.0.0-0 Loaded image: cortx-all:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-rgw:2.0.0-0 Loaded image: cortx-rgw:2.0.0-0 Loaded image: cortx-control:2.0.0-0 Loaded image: ip-172-31-39-22.ap-south-1.compute.internal:8080/seagate/cortx-control:2.0.0-0
> 
> @mukul-seagate11 Seems you are not using the local registry framework but copying images manually within the server.

@shailesh-vaidya, yes as this will be used in automation

**Review by shailesh-vaidya — APPROVED** [review]

File: solutions/community-deploy/CORTX-Deployment.md [hunk 1]
```diff
@@ -27,46 +31,60 @@ yum install git -y
...
+   - Then restart docker service on all the worker nodes by running, `systemctl restart docker`
+- To allow the PODs creation on primary node, pass the first input parameter for cluster-setup.sh script as true. Please note you must pass the input parameter as true for multi-node setup.
 ```
 ./cluster-setup.sh true
 ```
 
-## Deploy CORTX Stack 
-- Execute `cortx-deploy.sh` to deploy the CORTX stack on your K8s cluster.
+## Deploy CORTX Stack
+- Execute cortx-deploy.sh to deploy the CORTX stack on your K8s cluster.
```
Comment (gauravchaudhari02) [inline solutions/community-deploy/CORTX-Deployment.md:76]: Please add `` for cortx-deploy.sh

File: solutions/community-deploy/CORTX-Deployment.md [hunk 2]
```diff
@@ -27,46 +31,60 @@ yum install git -y
...
 ```
 export SOLUTION_CONFIG_TYPE=automated && ./cortx-deploy.sh --cortx-cluster
 ```
+- To execute `cortx-deploy.sh` on AWS to deploy the CORTX stack on your K8s cluster with locally generated images run the following command
```
Comment (gauravchaudhari02) [inline solutions/community-deploy/CORTX-Deployment.md:80]: can't we just say, `To execute cortx-deploy.sh on AWS cluster`?

**Review by shailesh-vaidya — APPROVED** [review]

**Pull request merged**